src/Yarhl/FileSystem/NodeFactory.cs

```diff
@@ -215,16 +216,89 @@ public static Node FromDirectory(string dirPath, string filter = "*", FileOpenMo
             bool subDirectories = false,
             FileOpenMode mode = FileOpenMode.ReadWrite)
         {
-            var options = subDirectories ?
+            if (string.IsNullOrEmpty(dirPath))
+                throw new ArgumentNullException(nameof(dirPath));
+
+            if (string.IsNullOrEmpty(filter))
+                throw new ArgumentNullException(nameof(filter));
+
+            if (string.IsNullOrEmpty(nodeName))
+                throw new ArgumentNullException(nameof(nodeName));
+
+            SearchOption options = subDirectories ?
+                SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+
+            // This sanitizes the path and remove double slashes
+            dirPath = Path.GetFullPath(dirPath);
+
+            string[] fileList = Directory.GetFiles(dirPath, filter, options);
+            return FromFileList(dirPath, nodeName, fileList, mode);
+        }
+
+        /// <summary>
+        /// Creates a Node containing all the files from the directory.
+        /// </summary>
+        /// <returns>The container node.</returns>
+        /// <param name="dirPath">Directory path.</param>
+        /// <param name="filter">Filter for files in directory.</param>
+        /// <param name="mode">The mode to open the files.</param>
+        public static Node FromDirectory(string dirPath, Func<string, bool> filter, FileOpenMode mode = FileOpenMode.ReadWrite)
+        {
+            if (string.IsNullOrEmpty(dirPath))
+                throw new ArgumentNullException(nameof(dirPath));
+
+            if (dirPath[dirPath.Length - 1] == Path.DirectorySeparatorChar)
```

Comment:
The one-line conditions not related to _guards_ should have braces to prevent future bugs (the previous function doesn't follow this rule this rule but I think it's old code before this rule was introduced).

Reply:
Done